packages/story-editor/src/components/colorPicker/basicColorPicker.js

```diff
@@ -76,58 +100,134 @@ function BasicColorPicker({
-              allowsGradient={allowsGradient}
-              aria-labelledby="colorpicker-saved-colors-title"
-            />
-          </>
-        )}
-        {allowsSavedColors && storyColors.length > 0 && (
-          <>
-            <Label id="colorpicker-story-colors-title">
-              {__('Story colors', 'web-stories')}
-            </Label>
-            <BasicColorList
-              color={color}
-              colors={storyColors}
-              handleColorChange={handleColorChange}
-              allowsOpacity={allowsOpacity}
-              allowsGradient={allowsGradient}
-              aria-labelledby="colorpicker-story-colors-title"
-            />
-          </>
-        )}
+        <EyedropperWrapper>
+          <Button
+            variant={BUTTON_VARIANTS.SQUARE}
+            type={BUTTON_TYPES.QUATERNARY}
+            size={BUTTON_SIZES.SMALL}
+            aria-label={__('Pick a color from canvas', 'web-stories')}
+            onClick={initEyedropper()}
+            onPointerEnter={initEyedropper(false)}
+          >
+            <Icons.Pipette />
+          </Button>
+          <StyledText size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+            {__('Sample color', 'web-stories')}
+          </StyledText>
+        </EyedropperWrapper>
+        <SavedColors>
+          {allowsSavedColors && (
+            <>
+              <Label id="colorpicker-story-colors-title">
+                {__('Current story', 'web-stories')}
+              </Label>
+              <BasicColorList
+                color={color}
+                colors={storyColors}
+                handleClick={handleClick}
+                colorType="local"
+                allowsOpacity={allowsOpacity}
+                allowsGradient={allowsGradient}
+                aria-labelledby="colorpicker-story-colors-title"
+                isEditMode={isEditMode}
+              />
+            </>
+          )}
+          {allowsSavedColors && (
+            <>
+              <Label id="colorpicker-saved-colors-title">
+                {__('Saved colors', 'web-stories')}
+              </Label>
+              <BasicColorList
+                color={color}
+                colors={savedColors}
+                colorType="global"
+                handleClick={handleClick}
+                allowsOpacity={allowsOpacity}
+                allowsGradient={allowsGradient}
+                aria-labelledby="colorpicker-saved-colors-title"
+                isEditMode={isEditMode}
+              />
+            </>
+          )}
+        </SavedColors>
+        <DefaultColors>
+          <Label id="colorpicker-default-colors-title">
+            {__('Default', 'web-stories')}
+          </Label>
+          <BasicColorList
+            color={color}
+            colors={BASIC_COLORS}
+            handleClick={(pattern) => {
+              handleColorChange(pattern);
+              if (isEditMode) {
+                setIsEditMode(false);
+              }
```

Comment:
Nit: If `isEditMode` is already false, setting it to false again is ignored (because it's a primitive), so for ease of implementation this `if` could be removed and you could just always set it to false here.